src/tcgwars/logic/impl/gen3/PopSeries2.groovy

```diff
@@ -370,7 +436,12 @@ public enum PopSeries2 implements LogicCardInfo {
         weakness P
         pokePower "Time Reversal", {
           text "Once during your turn, when you put Celebi ex from your hand onto your Bench, you may search your discard pile for a card, show it to your opponent, and put it on top of your deck."
-          actionA {
+          onActivate {r->
+            if (r==PLAY_FROM_HAND && bg.em().retrieveObject("Time_Reversal")!=bg.turnCount && my.discard.filterByType(TRAINER) && confirm('Use Time Reversal to search your discard pile for a card to put on top of your deck?')) {
+              bg.em().storeObject("Time_Reversal", bg.turnCount)
+              powerUsed()
+              my.discard.select("Select a card to put on top of your deck").showToOpponent("Selected card to move to the top of their deck").moveTo(addToTop:true, my.deck)
+            }
```

Comment:
Why is this being limited to 1 Time Reversal per turn? Should be reusable with super scoop up, as far as I understand at least. Also, it puts any card, not just trainers.

```suggestion
          onActivate {r->
            if (r==PLAY_FROM_HAND && my.discard && confirm('Use Time Reversal to search your discard pile for a card to put on top of your deck?')) {
              powerUsed()
              my.discard.select("Select a card to put on top of your deck").showToOpponent("Selected card to move to the top of their deck").moveTo(addToTop:true, my.deck)
            }
```